WinUIGallery/Controls/PageHeader.xaml.cs

```diff
@@ -98,5 +101,50 @@ private void UserControl_Loaded(object sender, RoutedEventArgs e)
         {
             APIDetailsBtn.Visibility = Visibility.Collapsed;
         }
+        if (Item != null)
+        {
+            FavoriteButton.IsChecked = StringListSettingsHelper.Contains(SettingsKeys.Favorites, Item.UniqueId);
+            StringListSettingsHelper.TryAddItem(SettingsKeys.RecentlyVisited, Item.UniqueId, InsertPosition.First, _maxRecentlyVisitedPages);
+        }
+    }
+
+    private string GetFavoriteGlyph(bool? isFavorite)
+    {
+        return (bool)isFavorite ? "\uE735" : "\uE734";
+    }
+
+    private string GetFavoriteToolTip(bool? isFavorite)
+    {
+        return (bool)isFavorite ? "Remove from favorites" : "Add to favorites";
+    }
+
+    private async void FavoriteButton_Click(object sender, RoutedEventArgs e)
+    {
+        if (sender is ToggleButton toggleButton && Item != null)
+        {
+            if (toggleButton.IsChecked == true)
+            {
+                if (!StringListSettingsHelper.TryAddItem(SettingsKeys.Favorites, Item.UniqueId, InsertPosition.Last, maxSize:12, trimEnabled:false))
```

Comment:
Why only 12? I do not think that we should limit this. On the topic of limiting, even if you add every page as favorite, the resulting string is less than 3KB so we do not need this check if we do not want to limit favorites.